### PR TITLE
Update emsdk dependency to match what emsdk produces

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Review by lewing — APPROVED** [review]

**Comment by lewing** [comment]
actually it looks there is a reference to this package in part of the tooling, let me see if I can clean that up

**Comment by mmitche** [comment]
@lewing I also needed to clean up the version props, if you aren't getting to that too

**Review by carlossanlop — APPROVED** [review]

File: src/mono/mono.proj [hunk 1]
```diff
@@ -85,7 +85,7 @@
     <PackageReference Condition="'$(HostOS)' == 'windows'"
                       Include="Microsoft.NET.Runtime.Emscripten.3.1.34.Python.win-x64"
```
Comment (lewing) [inline src/mono/mono.proj:86]:
```suggestion
                      Include="Microsoft.NET.Runtime.Emscripten.3.1.56.Python.win-x64"
```

File: src/mono/mono.proj [hunk 2]
```diff
@@ -259,7 +259,7 @@ JS_ENGINES = [NODE_JS]
     </PropertyGroup>
 
     <RemoveDir Directories="$(EMSDK_PATH)" />
-    
+
     <ItemGroup>
       <EmsdkFiles   Condition="'%(PackageReference.Identity)' != 'runtime.$(_portableHostOS)-$(BuildArchitecture).Microsoft.NETCore.Runtime.Wasm.Node.Transport' and '%(PackageReference.Identity)' != 'Microsoft.NET.Runtime.Emscripten.3.1.34.Python.win-$(BuildArchitecture)'"
```
Comment (lewing) [inline src/mono/mono.proj:264]:
```suggestion
      <EmsdkFiles   Condition="'%(PackageReference.Identity)' != 'runtime.$(_portableHostOS)-$(BuildArchitecture).Microsoft.NETCore.Runtime.Wasm.Node.Transport' and '%(PackageReference.Identity)' != 'Microsoft.NET.Runtime.Emscripten.$(EmskVersion).Python.win-$(BuildArchitecture)'"
```

File: src/mono/mono.proj [hunk 3]
```diff
@@ -108,7 +108,7 @@
     Only wasi doesn't come with its own python and windows uses 32-bit python, so refer to the transport pack instead.
   -->
   <PropertyGroup Condition="'$(HostOS)' == 'windows' and '$(TargetsWasi)' == 'true'">
-    <PythonCmd>$([MSBuild]::NormalizePath('$(PkgMicrosoft_NET_Runtime_Emscripten_3_1_34_Python_win-x64)', 'tools', 'python'))</PythonCmd>
+    <PythonCmd>$([MSBuild]::NormalizePath('$(PkgMicrosoft_NET_Runtime_Emscripten_3_1_56_Python_win-x64)', 'tools', 'python'))</PythonCmd>
```
Comment (lewing) [inline src/mono/mono.proj:111]: I don't see an obvious way to keep the version string out of this one, the plan is to move this logic to emsdk properly where it is easier to do

cc @radekdoulik

**Comment by lewing** [comment]
/ba-g analysis is green but preview branch

**Pull request merged**